packages/tfchain_client/test/contracts_test.dart

```diff
@@ -0,0 +1,164 @@
+      print(contracts);
+      expect(contracts, isNotEmpty);
+    });
+
+    test('Test Get Active Contracts by wrong node Id', () async {
+      try {
+        List<int> contracts =
+            await queryClient.contracts.getActiveContracts(nodeId: -21);
+        expect(contracts, isNotEmpty);
+      } catch (error) {
+        expect(error, isNotNull);
+      }
+    });
+
+    test('Test Get Contract Id by name', () async {
+      try {
+        BigInt? contractId =
+            await queryClient.contracts.getContractIdByName(name: "name");
+        expect(contractId, isNotEmpty);
+      } catch (error) {
+        expect(error, isNotNull);
+      }
+    });
+
+    test('Test Get Contract Id by node Id and hash', () async {
+      try {
+        BigInt? contractId = await queryClient.contracts
+            .getContractIdByNodeIdAndHash(nodeId: 21, hash: []);
+        expect(contractId, isNotEmpty);
+      } catch (error) {
+        expect(error, isNotNull);
+      }
+    });
+
+    test('Test Get Contract Lock by Id', () async {
+      try {
+        ContractLock? contractLock =
+            await queryClient.contracts.contractLock(id: BigInt.from(97897));
+        print(contractLock);
+        expect(contractLock, isNotEmpty);
+      } catch (error) {
+        expect(error, isNotNull);
+      }
+    });
+
+    test('Test Get a Dedicated Node extra fee', () async {
+      BigInt? fee =
+          await queryClient.contracts.getDedicatedNodeExtraFee(nodeId: 140);
+      print(fee);
+      expect(fee, isNotNull);
+    });
+  });
+
+  group("Test Contracts", () {
+    late Client client;
+    final mnemonic = Platform.environment['MNEMONIC']!;
+    final String link =
+        Platform.environment['LINK'] ?? 'wss://tfchain.dev.grid.tf/ws';
+    final String type = Platform.environment['SIGN_TYPE'] ?? 'sr25519';
+    BigInt? contractId = null;
+
+    setUp(() async {
+      client = Client(link, mnemonic, type);
+      await client.connect();
+    });
+
+    tearDownAll(() async {
+      if (contractId != null) {
+        try {
+          await client.contracts.cancel(contractId: contractId!);
+        } catch (error) {
+          print("Error canceling contract: $error");
+        }
+      }
+      await client.disconnect();
+    });
+
+    test('Test Create Node Contract with empty data', () async {
+      try {
+        BigInt? contractId = await client.contracts.createNode(
+            nodeId: 11, deploymentData: [], deploymentHash: [], publicIps: 0);
+      } catch (error) {
+        expect(error, isNotNull);
+      }
+    });
+
+    test('Test Update Node Contract with wrong data', () async {
+      try {
+        Contract? contract = await client.contracts
+            .updateNode(contractId: 11, deploymentData: [], deploymentHash: []);
+      } catch (error) {
+        expect(error, isNotNull);
+      }
+    });
+
+    test('Test Create Name Contract', () async {
+      contractId = await client.contracts.createName(name: "xxx");
```

Comment:
might be reserved by anyone else. you should generate a random name and ensure it's not reserved by anyone